service/src/test/java/com/ge/predix/acs/monitoring/AcsHealthAggregatorTest.java

```diff
@@ -0,0 +1,85 @@
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+// Modified version of org.springframework.boot.actuate.health.OrderedHealthAggregatorTests
+public class AcsHealthAggregatorTest {
+
+    private AcsHealthAggregator healthAggregator;
+
+    @BeforeMethod
+    public void beforeMethod() {
+        this.healthAggregator = new AcsHealthAggregator();
+    }
+
+    @Test
+    public void defaultOrder() {
+        Map<String, Health> healths = new HashMap<>();
+        healths.put("h1", new Health.Builder().status(Status.DOWN).build());
+        healths.put("h2", new Health.Builder().status(Status.UP).build());
+        healths.put("h3", new Health.Builder().status(Status.UNKNOWN).build());
+        healths.put("h4", new Health.Builder().status(Status.OUT_OF_SERVICE).build());
+        Assert.assertEquals(this.healthAggregator.aggregate(healths).getStatus(), Status.DOWN);
+    }
+
+    @Test
+    public void customOrder() {
+        this.healthAggregator.setStatusOrder(Status.UNKNOWN, Status.UP, Status.OUT_OF_SERVICE, Status.DOWN);
+        Map<String, Health> healths = new HashMap<>();
+        healths.put("h1", new Health.Builder().status(Status.DOWN).build());
+        healths.put("h2", new Health.Builder().status(Status.UP).build());
+        healths.put("h3", new Health.Builder().status(Status.UNKNOWN).build());
+        healths.put("h4", new Health.Builder().status(Status.OUT_OF_SERVICE).build());
+        Assert.assertEquals(this.healthAggregator.aggregate(healths).getStatus(), Status.UNKNOWN);
+    }
+
+    @Test
+    public void defaultOrderWithCustomStatus() {
+        Map<String, Health> healths = new HashMap<>();
+        healths.put("h1", new Health.Builder().status(Status.DOWN).build());
+        healths.put("h2", new Health.Builder().status(Status.UP).build());
+        healths.put("h3", new Health.Builder().status(Status.UNKNOWN).build());
+        healths.put("h4", new Health.Builder().status(Status.OUT_OF_SERVICE).build());
+        healths.put("h5", new Health.Builder().status(new Status("CUSTOM")).build());
+        Assert.assertEquals(this.healthAggregator.aggregate(healths).getStatus(), Status.DOWN);
+    }
+
+    @Test(dataProvider = "statuses")
+    public void defaultOrderWithDegradedStatus(final Status expectedStatus, final List<Status> actualStatuses) {
+        Map<String, Health> healths = new HashMap<>();
+        healths.put("h1", new Health.Builder().status(actualStatuses.get(0)).build());
+        healths.put("h2", new Health.Builder().status(actualStatuses.get(1)).build());
+        healths.put("h3", new Health.Builder().status(actualStatuses.get(2)).build());
+        Assert.assertEquals(this.healthAggregator.aggregate(healths).getStatus(), expectedStatus);
+    }
```

Comment:
Add a test case for:
healthAggregator.aggregate(empty healths); expected status is UKNOWN
healthAggregator.aggregate(healths contains one entry with 'cache' and status DOWN); expected status DEGRADED_STATUS
healthAggregator.aggregate(healths contains one entry with CUSTOM status); expected status UNKNOWN